content/design-patterns/read-receipts.textile

```diff
@@ -0,0 +1,63 @@
+</a>
+
+h2(#channels). Channels
+
+Let's name the three channels as below: 
+
+* "messages-channel"
+* "presence-channel"
+* "read-receipts-channel"
+
+Note that the arrows in the illustration above depict the direction of flow of data in these channels.
+
+The messages channel can be used by the users to send and receive messages from other users. Hence, they need to have both publish and subscribe permissions on this channel. This messages channel can be set up to use the "Reactor":https://www.ably.io/reactor service, so you get a buffer/ shock absorber to cope with a high volume of incoming messages. "Reactor": https://www.ably.io/reactor is a scalable service that lets you implement load balancing on your servers, trigger events or serverless functions in real time and even stream data in real time to other third-party streaming services.
+
+Next, the presence channel can be used by the users to send their "presence events":/realtime/presence#presence-states to the server and also receive presence events from other users via the server. The users have both publish as well as subscribe rights on this channel.
+
+The final one is the read-receipts channel. This channel is our main point of discussion throughout this Design Pattern. We'll use this channel to send and receive various message delivery receipts i.e, 'sent', 'delivered' and 'seen'. Everyone has publish as well as subscribe permissions to this channel. 
+
+h2(#order). Latencies are sorted but what about the order of messages?
+
+Ably ensures that messages are delivered to persistently connected subscribers on a channel in the order they were published on that channel by any other Ably client when using one of the Realtime libraries. Read more about it in the "support articles":https://support.ably.io/support/solutions/articles/3000044641-reliable-message-ordering-for-connected-clients.
+
+h2(#caveats). Caveats
+
+* Ably currently caps peak subscriptions to presence updates at 200, hence be mindful of that. If you don't need your users to know the presence status of all the other users, this problem is completely solved. In that case, the presence channel can be used by the clients to send their presence events to the server. In turn, the server can subscribe to this channel to be notified of this info whenever a presence event occurs. Everyone will have presence permissions on the presence channel, but not subscribe permission. This means you have the ability to query the presence on a channel or subscribe to presence events from your server, but the clients themselves won’t be able to inspect/subscribe to this channel. This avoids the classic n­-squared problem i.e. 10k people entering and subscribing to 10k ‘presence-enter’ events would result in 100m messages. However, 10k people entering, generating 10k presence events only for your server results in just 20k messages. 
```

Comment:
"i.e. 10k people entering and subscribing" => "e.g. 10k people...", as we're providing an example here.